tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -27,18 +29,29 @@ class DatalakeServices():
     def __init__(self,
                  test_ctx,
                  redpanda: RedpandaService,
-                 filesystem_catalog_mode=True,
                  include_query_engines: list[QueryEngineType] = [
                      QueryEngineType.SPARK, QueryEngineType.TRINO
-                 ]):
+                 ],
+                 catalog_type_and_impl: Tuple[CatalogType,
+                                              Optional[CatalogImpl]] = (
+                                                  CatalogType.REST,
+                                                  CatalogImpl.JDBC),
```

Comment:
Alluded to this in another comment, but different implementations with the REST wrapper IMO deserve to be declared different catalog types, unless you foresee CatalogImpl being useful for something else besides selecting between filesystem mode in the REST wrapper?